src/kernel/mpipeline.ml

```diff
@@ -325,3 +336,24 @@ let timing_information t = [
   "typer"  , !(t.typer_time);
   "error"  , !(t.error_time);
 ]
+
+let cache_information t =
+  let fmt_bool hit = if hit then "hit" else "miss" in
+  let typer_msg =
+    match !(t.typer_cache_stats) with
+    | Miss -> "miss"
+    | Hit { reused; typed } -> Printf.sprintf "reused: %d, typed: %d" reused typed in
+  let fmt_hit_miss h m = Printf.sprintf "hit: %d, miss: %d" h m in
```

Comment:
I think we should add more structure here for future tools parsing the cache stats:
Either: 

```
cmt : { 
  hit: int,
  miss: int,
}
```

or:

```
cmt_hist : int,
cmt_miss: int,
```

Reply:
done

Reply:
see PR description for updated json schema